internal/transform/ledger_test.go

```diff
@@ -0,0 +1,153 @@
+package transform
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stellar/stellar-etl/internal/utils"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/stellar/go/exp/ingest/ledgerbackend"
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/txnbuild"
+	"github.com/stellar/go/xdr"
+)
+
+func createTestArchiveBackend() *ledgerbackend.HistoryArchiveBackend {
+	archiveStellarURL := "http://history.stellar.org/prd/core-live/core_live_001"
+	backend, err := ledgerbackend.NewHistoryArchiveBackendFromURL(archiveStellarURL)
+	utils.PanicOnError(err)
+	return backend
+}
+func TestTransformOnTrueLedger(t *testing.T) {
+	sequenceNumber := uint32(30578981)
+	backend := createTestArchiveBackend()
```

Comment:
I should've made this clearer, but in unit testing, don't establish a network connection or get live data. This is generally because the file that lives at a specific URL can change, and introducing a live network connection adds a source of significant network latency to your testing. You'd like to be able to run the tests as quickly as possible.

I get the impetus in this case to add the extra test, but since you're also testing these functions with hand-crafted test cases below, the only additional information is whether or not the `ledgerbackend` struct is working as expected. But this information would be handled in a separate phase of the pipeline.